client/jetpack-cloud/sections/overview/foldable-nav/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { FoldableCard } from '@automattic/components';
+import './style.scss';
+import { useTranslate } from 'i18n-calypso';
+import { useCallback } from 'react';
+import { SidebarNavigatorMenuItem } from 'calypso/layout/sidebar-v2/navigator/navigator-menu-item';
+import { useDispatch, useSelector } from 'calypso/state';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { savePreference } from 'calypso/state/preferences/actions';
+import { getPreference } from 'calypso/state/preferences/selectors';
+import { Props } from './types';
+
+const FoldableNav = ( {
+	header,
+	navItems,
+	expandedInit,
+	compact,
+	preferenceName,
+	tracksName,
+}: Props ) => {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
+	const preference = useSelector( ( state ) => getPreference( state, preferenceName ) );
+	const expanded = preference?.expanded ?? expandedInit;
+
+	const savePreferenceType = useCallback(
+		( type: string, value: boolean ) => {
+			dispatch(
+				savePreference( preferenceName, {
+					...preference,
+					[ type ]: value,
+				} )
+			);
+		},
+		[ dispatch, preference, preferenceName ]
+	);
+
+	const onOpen = () => {
+		savePreferenceType( 'expanded', true );
+		if ( tracksName ) {
+			dispatch( recordTracksEvent( tracksName + '_expanded_true' ) );
```

Comment:
We refrain from creating event names programmatically since doing that hurts searchability. Let's say someone finds the event name via the browser's network tab and wants to know where and how it is triggered. Unfortunately, that person won't be able to find the event in the code base by searching for its name.

So, it these cases, it is expected to repeat ourselves.

FYI: @cleacos @coder-karen @gogdzl.

Reply:
The same applies to the arguments we pass to `translate`.

Reply:
> We refrain from creating event names programmatically since doing that hurts searchability.

Would you like us to change this now or just keep it in mind for future reference? I generally agree with this, and could easily adjust this to pass a param to the tracks event instead.

> Let's say someone finds the event name via the browser's network tab and wants to know where and how it is triggered. 

Are you also saying that we should define the tracks name in the same place as the trigger? If so, that makes things far more complicated.

> The same applies to the arguments we pass to `translate`.

Agreed; otherwise the text wouldn't be passed to the translator. Do you have any examples where we did otherwise?

Reply:
> Would you like us to change this now or just keep it in mind for future reference? I generally agree with this, and could easily adjust this to pass a param to the tracks event instead.

Yes. Let's change this as part of a follow-up task (you don't need to delay shipping the project).

> Are you also saying that we should define the tracks name in the same place as the trigger? If so, that makes things far more complicated.

That is not necessary. People will find their way if they find a match for the event name in the codebase. 

> Agreed; otherwise the text wouldn't be passed to the translator. Do you have any examples where we did otherwise?

I apologize for not being clear about this; I only mentioned it as an example. I didn't see any example of this as part of the implementation of the overview page.

Thanks for your questions!

Reply:
Great! Logged here: Automattic/jetpack-manage#201